### PR TITLE
feat: connect lobby to backend

### PR DESCRIPTION
**Pull request by thilan-tran**
## Screens
<p align="center">

</p>

## Changes
- load lobby information from backend
- clean up loading screens
- link to each song in Spotify

### CODE REVIEW

File: pages/index.tsx [hunk 1]
```diff
@@ -17,8 +22,11 @@ export default function Home(): JSX.Element {
       const token = indexCookie('token');
       if (!token || token === 'undefined') return;
 
-      const ids = await fetchAllLobbies(token);
-      const lobbyData = await Promise.all(ids.lobbies.map(id => fetchLobbyById(id, token)));
+      const res = await fetchAllLobbies(token);
+      const lobbyData = await Promise.all(res.lobbies.map(async id => {
+        const lobby = await fetchLobbyById(id, token);
+        return {...lobby, id};
+      }));
```
Comment (BryanPan342) [inline pages/index.tsx]: we dont need this anymore!

File: pages/index.tsx [hunk 2]
```diff
@@ -30,10 +38,14 @@ export default function Home(): JSX.Element {
 
     const data = await createLobby({lobbyName: name, theme}, token);
     const lobby = await fetchLobbyById(data.id, token);
-    setLobbies(lobbies.concat(lobby));
+    setLobbies(lobbies.concat({...lobby, id: data.id}));
```
Comment (BryanPan342) [inline pages/index.tsx:34]:
```suggestion
    setLobbies(prev => [...prev, {...lobby, id: data.id}]);
```

File: pages/index.tsx [hunk 3]
```diff
@@ -1,14 +1,19 @@
...
 import Link from '../components/Link';
+import Loading from '../components/Loading';
 import useModal from '../hooks/useModal';
 import {fetchAllLobbies, createLobby, fetchLobbyById} from '../utils/aurgy';
 import {indexCookie} from '../utils/cookies';
 import {ILobbyData} from '../utils/lobby-data';
 
+interface LobbyData extends ILobbyData {
+  id: string;
+}
```
Comment (BryanPan342) [inline pages/index.tsx]: i dont understand why u need id

File: pages/lobby.tsx [hunk 4]
```diff
@@ -1,100 +1,69 @@
...
   const [Modal, showModal, hideModal] = useModal();
-  const [lobbyData, setLobbyData] = useState<ILobbyData>(null);
-  const [error, setError] = useState(null);
+  const [lobbyData, setLobbyData] = useState<LobbyData>(null);
+  const [error, setError] = useState<string>(null);
+
+  function reloginUser() {
+    signOut();
+    void router.push(getUrlPath() + '/me');
+  }
```
Comment (BryanPan342) [inline pages/lobby.tsx]: smth tells me this should be part of the app context? like as part of sign out

**Review by BryanPan342 — DISMISSED** [review]
i think we need to reconnect

**Comment by thilan-tran** [comment]
## Changes
- update to work with API changes
- merge with #11 
- use shaders on home page
- add footer on home page
- set lobby text on a curve with [circletype.labwire.ca](https://circletype.labwire.ca/)
- TODO: change create button to a plus icon

File: utils/lobby-data.ts [hunk 5]
```diff
@@ -3,14 +3,33 @@ export type ILobbyCreationData = {
...
+  readonly lobbies: ILobbyDataShort[];
+}
+
+export type ILobbyDataFull = {
+  readonly name: string;
+  readonly theme: string;
+  readonly managerId: string;
+  readonly managerName: string;
+  readonly songs: ILobbySong[];
+  readonly users: ILobbyUser[];
```
Comment (BryanPan342) [inline utils/lobby-data.ts]: to me, it seems that we can condense some of this using some clever `interfaces`

```ts
export interface ILobbyDataBase {
  readonly id: string;
  readonly name: string;
}

export interface ILobbyUser extends ILobbyBase {}

export interface ILobbySong extends ILobbyBase {
  readonly artists: string[];
  readonly contributors: string[];
}

export interface ILobbyDataBase extends ILobbyBase {
  readonly theme: string;
}

export interface ILobbyData extends ILobbyDataBase {
  reaodnly managerId: string;
  readonly managerName: string;
  readonly songs: ILobbySong[];
  readonly user: ILobbyUser[];
}

```

File: styles/lobby.module.scss [hunk 6]
```diff
@@ -1,5 +1,6 @@
 @use 'variables.module' as *;
 @use 'mixins.module' as *;
+@use 'LobbyCircle.module' as *;
```
Comment (BryanPan342) [inline styles/lobby.module.scss]: oh damn.. okay i dont know how i didn't notice this..

preferably only the variables.modules file will be *.. everything else should follow modular CSS..

```ts
@use 'mixins.module' as mixin;
``` 

and I think we can put LobbyCircle variables into the variable CSS file!

File: styles/LobbyCircle.module.scss [hunk 7]
```diff
@@ -1,35 +1,44 @@
 @use 'variables.module' as *;
+@use 'mixins.module' as *;
```
Comment (BryanPan342) [inline styles/LobbyCircle.module.scss:2]:
```suggestion
@use 'mixins.module' as mixin;
```

File: styles/LobbyCircle.module.scss [hunk 8]
```diff
@@ -1,35 +1,44 @@
...
+
+$circle-size: 35vh;
+$circle-offset: 4.5vh;
```
Comment (BryanPan342) [inline styles/LobbyCircle.module.scss]: move to variables

File: components/LobbyCircle.tsx [hunk 9]
```diff
@@ -15,18 +20,58 @@ const CreateOptions2Style = {
...
+  useEffect(() => {
+    tlRef.current = makeRotationTimeline(circRef.current);
+  }, []);
+
+  useEffect(() => {
+    const circText = new CircleType(ref.current);
+
+    function setRadius() {
+      circText.radius(window.innerHeight * .35 / 2).dir(-1);
+    }
```
Comment (BryanPan342) [inline components/LobbyCircle.tsx]: why isnt this just the size of the element? we have a ref already to it

File: components/LobbyCircle.tsx [hunk 10]
```diff
@@ -15,18 +20,58 @@ const CreateOptions2Style = {
...
+    setRadius();
+    window.addEventListener('resize', setRadius);
 
+    return () => {
+      window.removeEventListener('resize', setRadius);
+      circText.destroy();
+    };
+  }, [ref]);
```
Comment (BryanPan342) [inline components/LobbyCircle.tsx]: why not just on mount?

File: components/LobbyCircle.tsx [hunk 11]
```diff
@@ -15,18 +20,58 @@ const CreateOptions2Style = {
...
+      onMouseEnter={animateForwards}
+      onMouseLeave={animateBackwards}
+    >
+      <PlaylistVisual title='' subtitle='' fullSize={false} animate={false} />
+      <div className={styles.name} ref={ref}>
+        {padName}
+      </div>
+    </div>
   );
+  return href ? (<NextLink href={href}>{base}</NextLink>) : base;
```
Comment (BryanPan342) [inline components/LobbyCircle.tsx]: Let's modify the `Link` component  that i already made to cover this feature too.. maybe like a `isDisabled` prop

File: components/Footer.tsx [hunk 12]
```diff
@@ -0,0 +1,12 @@
+import React from 'react';
+import styles from '../styles/Navbar.module.scss';
+
+export default function Footer(): JSX.Element {
+  return (
+    <div id={styles.navbar}>
```
Comment (BryanPan342) [inline components/Footer.tsx]: erm id?

File: pages/index.tsx [hunk 13]
```diff
@@ -1,33 +1,65 @@
...
-          <LobbyCircle name={lobby} key={index}/>
+        <LobbyCircle
+          name='CREATE'
+          onClick={() => showModal()}
+          create={!lobbies.length ? CreateOptions.CreateOnly : CreateOptions.CreateWithLobbies} />
+        {lobbies.map((lobby, index) => (
+          <LobbyCircle name={lobby.name} href={'/lobby?id=' + lobby.id} key={index} />
         ))}
       </div>
+      <Footer />
```
Comment (BryanPan342) [inline pages/index.tsx]: smth tells me this should be in the layout

File: pages/lobby.tsx [hunk 14]
```diff
@@ -107,25 +70,37 @@ function Lobby(): JSX.Element {
...
-              </div>
-            </div>
-          ))}
+          {lobbyData.songs.length
+            ? lobbyData.songs.map((song) => {
+              const artists = song.artists.join(', ');
+              const contributors = addShapeToProps(song.contributors);
+              return (
+                <div key={`${song.name}-${artists}`} className={styles.song}>
+                  <Tooltip text="PLAY">
```
Comment (BryanPan342) [inline pages/lobby.tsx]: let's remove this tooltip because we aren't planning on playing anytime soon

File: pages/lobby.tsx [hunk 15]
```diff
@@ -9,92 +10,54 @@ import useModal from '../hooks/useModal';
...
-          {error ?? 'LOADING...'}
-        </div>
-      </Layout>
+  if (!lobbyData) {
+    return <Loading error={error} />;
+  }
+
+  function addShapeToProps(users: string[]): NameplateProps[] {
+    return users.map(
+      name => ({name, shape: lobbyData.participantShapes[name] ?? 'pentagon'}),
```
Comment (BryanPan342) [inline pages/lobby.tsx]: nit: might be prudent just to keep a map of contributor to the prop values

**Review by BryanPan342 — DISMISSED** [review]
looking a lot more up to date! got a couple of things I would like you to change

**Comment by BryanPan342** [comment]
@thilan-tran what we could do is just take the repo you have and copy the files directly over

**Pull request closed without merging**